### PR TITLE
Backbone: replace all Function types

### PR DESCRIPTION
**Pull request by khasanovbi**
subj

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

File: types/backbone/index.d.ts [hunk 1]
```diff
@@ -515,7 +518,7 @@ declare namespace Backbone {
         decodeFragment(fragment: string): string;
         getSearch(): string;
         stop(): void;
-        route(route: string | RegExp, callback: Function): number;
+        route(route: string | RegExp, callback: (fragment: string) => void): number;
```
Comment (jgonggrijp) [inline types/backbone/index.d.ts:521]:
```suggestion
        route(route: string | RegExp, callback: RouterCallback): number;
```

**Review by jgonggrijp — CHANGES_REQUESTED** [review]
Excellent, thanks. Just one tiny oversight, see below.

File: types/backbone/index.d.ts (hunk 1 above)
Comment (khasanovbi) [reply types/backbone/index.d.ts:521]: `History` callback differs from `Router` callback. 

`route` is here: 
https://github.com/jashkenas/backbone/blob/master/backbone.js#L1933

Usage:
https://github.com/jashkenas/backbone/blob/master/backbone.js#L1961

Comment (jgonggrijp) [reply types/backbone/index.d.ts:521]: You are absolutely right, I stand corrected. This line also proves your point.

https://github.com/jashkenas/backbone/blob/153dc41616a1f2663e4a86b705fefd412ecb4a7a/backbone.js#L1676

**Review by jgonggrijp — APPROVED** [review]

**Comment by khasanovbi** [comment]
Ready to merge

**Pull request merged**